src/SideNav.js

```diff
@@ -1,21 +1,81 @@
-import React, { PropTypes } from 'react';
-import Icon from './Icon';
+import React, { Component, PropTypes } from 'react';
 import idgen from './idgen';
 
-const SideNav = ({ children, id = [`sidenav_${idgen()}`] }) =>
-  <nav>
-    <ul className='right hide-on-med-and-down'>{children}</ul>
-    <ul id={id} className='side-nav'>{children}</ul>
-    <a href='#' data-activates={id} className='button-collapse'>
-      <Icon>view_headline</Icon>
-    </a>
-  </nav>;
+class SideNav extends Component {
+  componentDidMount () {
+    const { options = {}, right, closeOnClick } = this.props;
+    $(this._trigger).sideNav({
+      ...options,
+      edge: right || options.edge,
+      closeOnClick: closeOnClick || options.closeOnClick
+    });
+  }
+
+  componentWillUnmount () {
+    $(this._trigger).sideNav('hide');
+  }
+
+  render () {
+    const { children, id = [`sidenav_${idgen()}`], ...props } = this.props;
+    this.id = id;
```

Comment:
True, wouldn't look good, your call to either add it in constructor or inline :)